### DISCUSSION TITLE
Launching Terminal.Gui 1.0

### DISCUSSION THREAD

**Discussion by tig** [tui-cs/Terminal.Gui#1060] [General]
Moving this Issue to discussions:

https://github.com/migueldeicaza/gui.cs/issues/612

These are the current bugs filed against 1.0:

https://github.com/migueldeicaza/gui.cs/labels/1.0

We need to focus on closing these bugs in order to release 1.0. I do not think we should accept ANY new features or capabilities (including TreeView and TableView) until after we get 1.0 out. Does anyone disagree with this?

**Comment by BDisp** [comment]
I think there are many users expecting this news features (`TreeView `and `TableView`), although needed mouse support.

**Reply by tig** [reply]
Sorry, but the intent of your message does not translate clearly. I think you are saying:

1) We should wait until TreeView and TableView are done and completely tested before launching 1.0.
2) We should wait until Windows Terminal supports mouse.

Does this capture what you meant to say?

Note that launching 1.0 doesn't mean we can't launch a 1.1 shortly there after with TreeView and TableView.

**Comment by BDisp** [comment]
Ahh ok. Sorry. If it don't take to long to launch a 1.1 shortly thereafter with `TreeView `and `TableView`, so there is no problem to launch the 1.0 before. The `Windows Terminal` unsupports mouse must not avoid to launch the 1.0 version.

**Comment by tig** [comment]
Ok team!!!! I think we have our first 1.0 candidate.

v1.0.0-beta.8 is now published to nuget.

Let's get the word out to folks to try it out and let it stew for a few days. Then we can declare 1.0 DONE and move on.

Woot!

**Comment by tig** [comment]
v1.0.0-beta.11 is now being pushed to nuget as a pre-release.

      * NEW CONTROL: Tabview - thanks @tznind!
      * UI Catalog now shows correct Terminal.gui.dll version
      * Fixes #939 - README sample does not compile - thanks @buzzfrog!
      * Fixes #1154 - FileDialog blank constructor results in unstable window
      * Fixes #1155 - MoveForward/MoveBackward not bound on Text controls.
      * Fixes #1152 - Generic TreeView`1 breaks upon selection in All Views Tester)
      * Fixes #1148 - TextFormatter.Format does not keep the end spaces on wrapped text.
      * Fixes #134 - (HUGE) TextView: Add line wrapping.
      * Fixes #1145 - ScrollBar down arrow is not showing.
      * Fixes #1143 - Cannot change the MenuBar's background color.
      * Fixes #1141 - Cannot change the StatusBar's background color.
      * Fixes #1048 - BrighCyan to BrightCyan spelling
      * Fixes #1130 - Broken Table/TreeView links in docs.

**Comment by tig** [comment]
v1.0.0-rc.1 is now being pushed to nuget as the first (only?!??!) release candidate.

One bug remains, which I cannot repro:

https://github.com/migueldeicaza/gui.cs/issues/931

Once we confirm this bug is fixed, I will accept PR #1227 and push the `v1.0.0` tag which will cause the final nuget package to get generated and published.

**Comment by tig** [comment]
Update:  v1.0.0-rc.7 is now on nuget.

`main` is updated to all the recent changes, including the code coverage (note code coverage badge in `README.md`.

We are still waiting for #931 to be fixed before launching 1.0.

**Reply by BDisp** [reply]
> We are still waiting for #931 to be fixed before launching 1.0.

I ended up finally fixed it.

**Comment by tig** [comment]
Update: v1.0.0-rc.9 is now on nuget.

`main` is updated.

We believe #931 is fixed. Let's get some folks who reported the problem previously to verify.

**Comment by tig** [comment]
I have pushed v1.0.0-rc.10 to nuget. This includes some more tweaks regarding #931 / #418.

I have discovered a regression in how keys are handled relative to `StatusBar`. This regression breaks `OutConsoleGridView` and I believe it needs to be fixed before we can launch 1.0. See #1256.

**Comment by tig** [comment]
I just pushed v1.0.0-rc.11 to nuget. This includes the fix to #1256. 

I will now retest `ocgv` to ensure it works with the published package....

**Comment by tig** [comment]
Ok, I have verified `ocgv` works great with the latest RC.

We will launch v1.0 when this is fixed:

https://github.com/migueldeicaza/gui.cs/issues/1255

**Comment by tig** [comment]
I just pushed v1.0.0-rc.13.

I think we're done! 

If anyone has any objections raise them now. I will push 1.0 at 12PM PT today (4/28/2021) if there are no objections.

**Comment by migueldeicaza** [comment]
Congratulations!   This is a thing of beauty!

Also - very happy that graphics made it into the release!   That is beautiful!
